sympy/vector/operators.py

```diff
@@ -250,24 +251,52 @@ def gradient(scalar_field, coord_sys=None, doit=True):
     10*R.x*R.z*R.i + 5*R.x**2*R.k
 
     """
-    coord_sys = _get_coord_sys_from_expr(scalar_field, coord_sys)
-
-    if coord_sys is None:
-        return Vector.zero
-    else:
-        from sympy.vector.functions import express
-        h1, h2, h3 = coord_sys.lame_coefficients()
-        i, j, k = coord_sys.base_vectors()
-        x, y, z = coord_sys.base_scalars()
-        scalar_field = express(scalar_field, coord_sys, variables=True)
-        vx = Derivative(scalar_field, x) / h1
-        vy = Derivative(scalar_field, y) / h2
-        vz = Derivative(scalar_field, z) / h3
+    from sympy.vector.functions import express, split_coordinate
+
+    def grad(expr):
```

Comment:
I wanted to calculate `gradient` for small piece of of expression, which depends on one coordinate system.